py2store/persisters/ssh_persister.py

```diff
@@ -0,0 +1,127 @@
+from collections.abc import MutableMapping
+from paramiko import *
+import os.path
+import stat
+
+
+def remote_mkdir(sftp, remote_directory):
+    """
+    Change to this directory, recursively making new folders if needed.
+    returns: True if any folders were created.
+    """
+    if remote_directory == '/':
+        # absolute path so change directory to root
+        sftp.chdir('/')
+        return False
+    if remote_directory == '':
+        # top-level relative directory must exist
+        return False
+    try:
+        sftp.chdir(remote_directory) # sub-directory exists
+    except IOError:
+        dirname, basename = os.path.split(remote_directory.rstrip('/'))
```

Comment:
Are you sure you need `.rstrip` here?

Reply:
remote_directory.rstrip('/') allows to avoid FileNotFoundError if rootdir ends by '/' (e.g. rootdir =  ./py2store3/)